src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/ObjectGetTypeDataflow.cs

```diff
@@ -295,5 +289,92 @@ public static void Test ()
 				TestMethodGenericParameterAsReturnType ();
 			}
 		}
+
+		class InstantiatedTypeParameterAsSource
+		{
+			[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)]
+			class Annotated {}
+
+			static T GenericReturnType<T> () => default;
+
+			static void TestGenericMethodReturnType ()
+			{
+				GenericReturnType<Annotated> ().GetType ().RequiresPublicFields ();
+			}
+
+			static void GenericOutParameter<T> (out T value) => value = default;
+
+			[UnexpectedWarning ("IL2072", Tool.Analyzer, "https://github.com/dotnet/runtime/issues/101734")]
+			static void TestGenericMethodOutParameter ()
+			{
+				GenericOutParameter (out Annotated value);
+				value.GetType ().RequiresPublicFields ();
+			}
+
+			class Generic<T>
+			{
+				public static T ReturnType () => default;
+
+				public static void OutParameter (out T value) => value = default;
+
+				public static T field;
+
+				public static T Property { get; }
+			}
+
+			static void TestGenericClassReturnType ()
+			{
+				Generic<Annotated>.ReturnType ().GetType ().RequiresPublicFields ();
+			}
+
+			[UnexpectedWarning ("IL2072", Tool.Analyzer, "https://github.com/dotnet/runtime/issues/101734")]
+			static void TestGenericClassOutParameter ()
+			{
+				Generic<Annotated>.OutParameter (out var value);
+				value.GetType ().RequiresPublicFields ();
+			}
+
+			static void TestGenericClassField ()
+			{
+				Generic<Annotated>.field.GetType ().RequiresPublicFields ();
+			}
+
+			static void TestGenericClassProperty ()
+			{
+				Generic<Annotated>.Property.GetType ().RequiresPublicFields ();
+			}
```

Comment:
Could we add another version of this but where the members are on a class nested in a generic class? Along the lines of this:

```C#
            class Generic<T>
			{
                class Nested
                {
				    public static T ReturnType () => default;

				    public static void OutParameter (out T value) => value = default;

				    public static T field;

				    public static T Property { get; }
                }
			}

			static void TestGenericClassReturnType ()
			{
				Generic<Annotated>.Nested.ReturnType ().GetType ().RequiresPublicFields ();
			}

			[UnexpectedWarning ("IL2072", Tool.Analyzer, "https://github.com/dotnet/runtime/issues/101734")]
			static void TestGenericClassOutParameter ()
			{
				Generic<Annotated>.Nested.OutParameter (out var value);
				value.GetType ().RequiresPublicFields ();
			}

			static void TestGenericClassField ()
			{
				Generic<Annotated>.Nested.field.GetType ().RequiresPublicFields ();
			}

			static void TestGenericClassProperty ()
			{
				Generic<Annotated>.Nested.Property.GetType ().RequiresPublicFields ();
			}
```